src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -0,0 +1,304 @@
+                                                                  log_file_prefix='shifterimg_pull')
+
+    with open(sp_std_out_file, "r") as std_out, open(sp_std_err_file, "r") as std_err:
+        sp_std_out = std_out.read()
+
+        if 'FAILURE' in sp_std_out:
+            raise ValueError(f"Error pulling Shifter image {image_str}.\n"
+                             f"Standard output: {sp_std_out}\n"
+                             f"Standard error: {std_err.read()}")
+
+
+def _fetch_image(registry, image_name, job_dir, tag='latest', force_pull=True):
+    """
+    Fetches the specified Shifter image if it is not already present on the system.
+
+    When force_pull is set to True, the image is always pulled from the registry
+    """
+    image_str = f"{registry}/{image_name}:{tag}"
+
+    if not force_pull:
+        # Check if the image is already present on the system
+        si_std_out_file, si_std_err_file, si_exit_code = _run_command(["shifterimg", "images"], job_dir,
+                                                                      log_file_prefix='shifterimg_images')
+
+        with open(si_std_out_file, "r") as f:
+            si_std_out = f.read()
+
+        images = si_std_out.split("\n")
+        for image in images:
+            parts = image.split()
+            if len(parts) != 6:
+                continue
+            if parts[5] == image_str:
+                print(f"Shifter image {image_name}:{tag} from registry {registry} already exists.")
+                return parts[5]
+
+    # Pull the image from the registry
+    _pull_image(image_str, job_dir)
+
+    return image_str
+
+
+def _create_shifter_wrapper(job_dir, image_str):
+    """
+    Creates the Shifter wrapper script.
+    """
+
+    # The content of the Shifter wrapper script
+    shifter_wrapper = f'''#!/bin/bash
+
+image={image_str}
+
+if [ $# -lt 1 ]; then
+    echo "Error: Missing command argument."
+    echo "Usage: shifter_wrapper.sh your-command-arguments"
+    exit 1
+fi
+
+command="$@"
+echo "Running shifter --image=$image $command"
+
+cd {job_dir}
+shifter --image=$image $command
+            '''
+
+    wrapper_file = os.path.join(job_dir, 'shifter_wrapper.sh')
+    with open(wrapper_file, "w") as f:
+        f.write(shifter_wrapper)
+
+    os.chmod(wrapper_file, 0o777)
+
+    return wrapper_file
+
+
+def _create_genome_id_file(genome_ids, genome_id_file):
+    """
+    Create a tab-separated values (TSV) file with a list of genome IDs.
+    """
+    with open(genome_id_file, 'w') as f:
+        f.write("genome_id\n")
+        for genome_id in genome_ids:
+            f.write(f"{genome_id}\n")
+
+
+def _create_task_list(source_data_dir, kbase_collection, load_ver, tool, wrapper_file, job_dir, root_dir,
+                      threads=256, program_threads=256, source_file_ext='genomic.fna.gz'):
+    """
+    Create task list file (tasks.txt)
+    """
+    genome_ids = [path for path in os.listdir(source_data_dir) if
```

Comment:
We'll probably be running things that aren't genomes in the future but we can cross that bridge when we come to it